framework/wazuh/rootcheck.py

```diff
@@ -29,20 +29,20 @@ def run(agent_list=None):
     result = AffectedItemsWazuhResult(all_msg='Rootcheck scan was restarted on returned agents',
                                       some_msg='Rootcheck scan was not restarted on some agents',
                                       none_msg='No rootcheck scan was restarted')
+
+    wq = WazuhQueue(common.ARQUEUE)
     for agent_id in agent_list:
         try:
-            agent_info = Agent(agent_id).get_basic_information()
-            agent_status = agent_info.get('status', 'N/A')
-            if agent_status.lower() != 'active':
-                result.add_failed_item(
-                    id_=agent_id, error=WazuhError(1601, extra_message='Status - {}'.format(agent_status)))
-            else:
-                wq = WazuhQueue(common.ARQUEUE)
+            agent_status = Agent(agent_id).get_basic_information().get('status', 'N/A')
```

Comment:
```suggestion
            agent = Agent(agent_id)
            agent.load_info_from_db()
            agent_status = agent.status if agent.status else 'N/A'
```

After testing it, this block of code I suggested is slightly faster than the one changed.
Avoid using `get_basic_information` as it will create a JSON using a lot of functions. We use this JSON to get the status but in truth, we can get the status after loading the agent information from the db with `load_info_from_db`.

Reply:
Update function typing too.